integration_tests/external/test_text_generation_no_mock.py

```diff
@@ -0,0 +1,266 @@
+    """If a turtle egg was kept warm, it would likely hatch into a baby turtle. The sex of the baby turtle would be determined by the incubation temperature, assuming the species is one of those that determine sex thermally. This is because many turtle species have the ability to move around inside their eggs to select the best temperature for development, which can influence their sexual destiny.""",
+]
+
+
+REFERENCES = [
+    """John Adams and Alexander Hamilton did not get along. John Adams had grown independent of his cabinet, often making decisions despite opposition from it.\n""",  # same as prediction with some strings deleted
+    """Yes, Lincoln won the election of 1860. He received the highest number of votes and a majority in the Electoral College, making him the 16th President of the United States. However, it's important to note that he won entirely due to his support in the North and West, as he did not receive any votes in 10 of the 15 Southern slave states.""",  # same as prediction
+    """If kept warm, it would hatch a coyote.""",  # very different than prediction
+]
+
+
+def test_llm_evaluation_with_openai(
+    client: Client,
+    gt_questions: list[GroundTruth],
+    pred_answers: list[Prediction],
+    dataset_name: str,
+    model_name: str,
+):
+    dataset = Dataset.create(dataset_name)
+    model = Model.create(model_name)
+
+    for gt in gt_questions:
+        dataset.add_groundtruth(gt)
+
+    dataset.finalize()
+
+    for pred in pred_answers:
+        model.add_prediction(dataset, pred)
+
+    model.finalize_inferences(dataset)
+
+    metrics_to_return = [
+        MetricType.AnswerRelevance,
+        MetricType.BLEU,
+        MetricType.Coherence,
+        MetricType.ROUGE,
+    ]
+
+    eval_job = model.evaluate_text_generation(
+        datasets=dataset,
+        metrics_to_return=metrics_to_return,
+        llm_api_params={
+            "client": "openai",
+            "data": {
+                "model": "gpt-4o",
+            },
+        },
+    )
+
+    assert eval_job.id
+    eval_job.wait_for_completion(timeout=90)
+
+    assert eval_job.wait_for_completion(timeout=30) == EvaluationStatus.DONE
+
+    metrics = eval_job.metrics
+
+    # Check that the right number of metrics are returned.
+    assert len(metrics) == len(pred_answers) * len(metrics_to_return)
+
+    # Check that the returned metrics have the right format.
+    for m in metrics:
+        if m["type"] in ["AnswerRelevance", "BLEU"]:
+            assert 0 <= m["value"] <= 1
+        if m["type"] == "Coherence":
+            assert m["value"] in {1, 2, 3, 4, 5}
+        if m["type"] == "ROUGE":
+            assert isinstance(m["value"], dict)
+            assert all(0 <= v <= 1 for v in m["value"].values())
```

Comment:
can you check the values of what's returned? i.e., coherence value should be >= 4, for example

Reply:
I broke this into two functions each, with data more curated to getting a good/bad metric score.